lib/ex_aws/cloudfront/custom_policy.ex

```diff
@@ -0,0 +1,125 @@
+defmodule ExAws.CloudFront.CustomPolicy do
+  @enforce_keys [:url, :date_less_than]
+
+  defstruct [:url, :date_less_than, :date_greater_than, :ip_address]
+
+  @doc """
+  Create a Custom Policy.
+  """
+  def create(url), do: create(url, ExAws.Utils.now_in_seconds + 1800)
+  def create(url, %DateTime{} = date_less_than), do: create(url, DateTime.to_unix(date_less_than))
+  def create(url, date_less_than) when is_binary(url) and is_integer(date_less_than) do
+    %__MODULE__{
+      url: url,
+      date_less_than: date_less_than
+    }
+  end
+
+  @doc """
+  Puts a beginning date and time in Unix time format and UTC.
+  """
+  def put_date_greater_than(%__MODULE__{} = policy, %DateTime{} = value) do
+    put_date_greater_than(policy, DateTime.to_unix(value))
+  end
+
+  def put_date_greater_than(%__MODULE__{} = policy, value) when is_nil(value) or is_integer(value) do
+    %__MODULE__{ policy | date_greater_than: value }
+  end
+
+  @doc """
+  Puts an IP address.
+  """
+  def put_ip_address(%__MODULE__{} = policy, value) when is_nil(value) or is_binary(value) do
+    %__MODULE__{ policy | ip_address: value }
+  end
```

Comment:
I'm not sure that this setter pattern is the best API. It seems like all of this could be simply:
```
CustomPolicy.new(url, opts)
```